src/CountVisitor.h

```diff
@@ -30,23 +30,23 @@ class CountVisitor {
   }
 
 public:
-  bool node(SourceNode<CentroidData> from, TargetNode<CentroidData> on) {
-    if (from.data.count == 0 || on.data->count == 0) {
+  bool node(const SpatialNode<CentroidData>& from, SpatialNode<CentroidData>& on) {
```

Comment:
Should `on` also have the const qualifier?

Reply:
on needs to be edited. for example, in GravityVisitor we call target.applyForce